pkg/il/compiler/compiler.go

```diff
@@ -31,18 +31,24 @@ type generator struct {
 	err     error
 }
 
+// CompilationResult is returned as the result of compilation.
+type CompilationResult struct {
```

Comment:
Just `Result`, you're already in `package compiler`, this is references as `compiler.CompilationResult` which is redundant; `compiler.Result` works just as well/more idiomatic. Plus the variable name that you store this into is always called `result`, which is another signal that the type can by named `Result` w/o losing anything.